src/raftstore/store/snap.rs

```diff
@@ -720,27 +1696,50 @@ impl SnapManagerCore {
         self.registry.contains_key(key)
     }
 
-    pub fn get_snapshot_to_build(&self, key: &SnapKey) -> io::Result<Box<Snapshot>> {
-        let f = try!(v1::Snap::new_for_writing(&self.base, self.snap_size.clone(), true, key));
-        Ok(Box::new(f))
+    pub fn get_snapshot_for_building(&self, key: &SnapKey) -> io::Result<Box<Snapshot>> {
+        if self.use_sst_file_snapshot {
+            let f = try!(v2::Snap::new_for_building(&self.base, key, self.snap_size.clone()));
+            Ok(Box::new(f))
+        } else {
+            let f = try!(v1::Snap::new_for_writing(&self.base, self.snap_size.clone(), true, key));
+            Ok(Box::new(f))
+        }
     }
 
-    pub fn get_snapshot_for_sending(&self, key: &SnapKey) -> io::Result<Box<Snapshot>> {
-        let reader = try!(v1::Snap::new_for_reading(&self.base, self.snap_size.clone(), true, key));
-        Ok(Box::new(reader))
+    pub fn get_snapshot_for_sending(&self, key: &SnapKey) -> RaftStoreResult<Box<Snapshot>> {
+        let s = try!(v1::Snap::new_for_reading(&self.base, self.snap_size.clone(), true, key));
+        if s.exists() {
+            return Ok(Box::new(s));
+        }
+        let s = try!(v2::Snap::new_for_sending(&self.base, key, self.snap_size.clone()));
+        Ok(Box::new(s))
     }
 
     pub fn get_snapshot_for_receiving(&self,
                                       key: &SnapKey,
-                                      _data: &[u8])
-                                      -> io::Result<Box<Snapshot>> {
-        let f = try!(v1::Snap::new_for_writing(&self.base, self.snap_size.clone(), false, key));
-        Ok(Box::new(f))
+                                      data: &[u8])
+                                      -> RaftStoreResult<Box<Snapshot>> {
+        let mut snapshot_data = RaftSnapshotData::new();
+        try!(snapshot_data.merge_from_bytes(data));
+        if snapshot_data.get_data().len() == 0 {
+            let f = try!(v1::Snap::new_for_writing(&self.base, self.snap_size.clone(), false, key));
+            Ok(Box::new(f))
+        } else {
+            let f = try!(v2::Snap::new_for_receiving(&self.base,
+                                                     key,
+                                                     snapshot_data,
+                                                     self.snap_size.clone()));
+            Ok(Box::new(f))
+        }
     }
 
-    pub fn get_snapshot_for_applying(&self, key: &SnapKey) -> io::Result<Box<Snapshot>> {
-        let f = try!(v1::Snap::new_for_reading(&self.base, self.snap_size.clone(), false, key));
-        Ok(Box::new(f))
+    pub fn get_snapshot_for_applying(&self, key: &SnapKey) -> RaftStoreResult<Box<Snapshot>> {
+        let s = try!(v1::Snap::new_for_reading(&self.base, self.snap_size.clone(), false, key));
```

Comment:
Same thing. The v2 version could always be checked prior to the v1 version, even when the `use_sst_file_snapshot` option is disabled.

Since neither of the v1 version or the v1/v2 mixed version live long in the code base, I don't think this priority make much sense.